benchmark/src/main/java/com/datatorrent/benchmark/monitor/SystemResourceMonitor.java

```diff
@@ -0,0 +1,171 @@
+
+    /**
+     * cpu
+     */
+    ProcessCpuTime, SystemCpuLoad, ProcessCpuLoad,
+
+    /**
+     * file
+     */
+    OpenFileDescriptorCount, MaxFileDescriptorCount
+  }
+
+  public static SystemResource[] memoryResources = new SystemResource[] {
+      SystemResource.CommittedVirtualMemorySize,
+      SystemResource.FreePhysicalMemorySize,
+      SystemResource.TotalPhysicalMemorySize,
+      SystemResource.TotalSwapSpaceSize,
+      SystemResource.FreeSwapSpaceSize};
+
+  public static SystemResource[] cpuResources = new SystemResource[] {
+      SystemResource.ProcessCpuTime,
+      SystemResource.SystemCpuLoad,
+      SystemResource.ProcessCpuLoad};
+
+  public static SystemResource[] fileResources = new SystemResource[] {
+      SystemResource.OpenFileDescriptorCount,
+      SystemResource.MaxFileDescriptorCount};
+
+  public static SystemResource[] allResources = SystemResource.values();
+
+  private static Map<Set<SystemResource>, SystemResourceMonitor> resourcesToMonitors = Maps.newHashMap();
+
+  private static SystemResourceMonitor defaultInstance;
+
+  public static SystemResourceMonitor create()
+  {
+    if (defaultInstance == null) {
+      synchronized (SystemResourceMonitor.class) {
+        if (defaultInstance == null) {
+          defaultInstance = new SystemResourceMonitor();
+        }
+      }
+    }
+    return defaultInstance;
+  }
+
+  public static SystemResourceMonitor create(SystemResource[] systemResources)
+  {
+    SystemResourceMonitor monitor = resourcesToMonitors.get(Sets.newHashSet(systemResources));
+    if (monitor == null) {
+      synchronized (SystemResourceMonitor.class) {
+        if (monitor == null) {
+          monitor = new SystemResourceMonitor(systemResources);
+          resourcesToMonitors.put(Sets.newHashSet(systemResources), monitor);
+        }
+      }
+    }
+    return monitor;
+  }
+
+  private SystemResource[] systemResources;
+  private OperatingSystemMXBean operatingSystemMXBean = ManagementFactory.getOperatingSystemMXBean();
+  private Map<SystemResource, Number> systemResourceUsage = Maps.newEnumMap(SystemResource.class);
+  private List<GarbageCollectorMXBean> garbageCollectorMXBeans = ManagementFactory.getGarbageCollectorMXBeans();
+
+  private SystemResourceMonitor()
+  {
+  }
+
+  private SystemResourceMonitor(SystemResource[] systemResources)
+  {
+    if (systemResources == null || systemResources.length == 0) {
+      throw new IllegalArgumentException("Resources should not null or empty");
+    }
+    this.systemResources = systemResources;
+  }
+
+  public Map<SystemResource, Number> getSystemResourceUsage()
+  {
+    systemResourceUsage.clear();
+    for (SystemResource sr : systemResources) {
+      Method getterMethod = getGetterMethod(sr);
+      getterMethod.setAccessible(true);
+      try {
+        systemResourceUsage.put(sr, (Number)getterMethod.invoke(operatingSystemMXBean));
+      } catch (Exception e) {
+        Throwables.propagate(e);
```

Comment:
should be
```java
 throw Throwables.propagate(e);
```
so that both IDE and the compiler know the execution flow stops here